lib/p2p/Pool.ts

```diff
@@ -57,6 +59,34 @@ class Pool extends EventEmitter {
 
     this.handshakeData = handshakeData;
 
+    if (!this.config.externalip) {
```

Comment:
It gets the external IP if you don't specify in the config with `p2p.externalip`. If you specify it doesn't try to find your external IP.

Reply:
My opinion is that we should leave it to the user to provide the external IP if he wants to be connected (inbound).

Reply:
We have an option for whether the user wants to listen to incoming peer connections or not. I figure it might make sense to use the configured listen port and our discovered external IP by default when listening is enabled and no external address is specified.

Reply:
I'm with @sangaman and @michael1011 . I think it should be automated in this way for convenience, I shouldn't have to check my external IP and put it in a config manually for other 'xud's being able to connect to me. Also this gives us a tool at hand to automatically check for a changing IP and update accordingly (https://github.com/ExchangeUnion/xud/issues/314) @offerm